scripts/generate-entry-i10n.js

```diff
@@ -0,0 +1,187 @@
+      return null;
+    })
+    .catch(function(error) {
+      console.log(error);
+    });
+}
+
+// function getThings(type) {
+//   db.any(`SELECT * FROM things WHERE type = '${type}'`)
+//     .then(function(thingData) {
+//       thingData.forEach(data => {
+//         getLocalizationData(data.id, data.original_language);
+//       });
+//       return null;
+//     })
+//     .catch(function(error) {
+//       console.log(error);
+//     });
+// }
+
+function getLocalizationData(thingid, language) {
+  db.any(
+    `SELECT * FROM localized_texts WHERE thingid = ${thingid} AND language = '${language}' ORDER BY timestamp DESC LIMIT 1`
+  )
+    .then(function(data) {
+      data.forEach(data => {
+        createNewRecord(data, thingid);
+      });
+      return null;
+    })
+    .catch(function(error) {
+      console.log(error);
+    });
+}
+
+async function createNewRecord(data, thingid) {
+  var currentLanguages = [];
+  await db
+    .any(`SELECT language FROM localized_texts WHERE thingid = ${thingid}`)
+    .then(function(existingLanguages) {
+      existingLanguages.forEach(lang => {
+        currentLanguages.push(lang.language);
+      });
+    })
+    .catch(function(error) {
+      console.log(error);
+    });
+
+  for (var i = 0; i < SUPPORTED_LANGUAGES.length; i++) {
+    // Loop in supported languages
+    const language = SUPPORTED_LANGUAGES[i];
+
+    if (currentLanguages.indexOf(language.twoLetterCode) < 0) {
+      // If language in loop not exist from currentLanguages. Add record
+      if (language.twoLetterCode !== data.language) {
+        // Don't create language from original
+        const item = {
+          body: "",
+          title: "",
+          description: "",
+          language: language.twoLetterCode,
+          thingid: thingid,
+        };
+
+        if (data.body) {
+          item.body = await translateText(data.body, language.twoLetterCode);
+        }
+
+        if (data.title) {
+          item.title = await translateText(data.title, language.twoLetterCode);
+        }
+
+        if (data.description) {
+          item.description = await translateText(
+            data.description,
+            language.twoLetterCode
+          );
+        }
+
+        console.log(
+          `ThingID: ${thingid} => saving record for ${
+            language.twoLetterCode
+          } from ${data.language}`
+        );
+        // await saveRecord([item]);
+        console.log(
+          `ThingID: ${thingid} => record for ${language.twoLetterCode} from ${
+            data.language
+          } DONE!`
+        );
+        console.log(`ThingID: ${item.title}`);
+        console.log(
+          "====================================================================="
+        );
+      }
+    }
+  }
+  console.log(`Data length total: ${dataLengthTotal}`);
+}
+
+async function saveRecord(records) {
+  const insert = pgp.helpers.insert(
+    records,
+    ["body", "title", "description", "language", "thingid"],
```

Comment:
we should  add the timestamp data when we save a record (the field already exists in the table)

Reply:
done